includes/class-newspack-newsletters-subscription.php

```diff
@@ -489,6 +554,23 @@ public static function process_subscription_intents( $intent_id = null ) {
 				if ( $user ) {
 					delete_user_meta( $user->ID, 'newspack_newsletters_subscription_error' );
 				}
+
+				/**
+				 * Fires after a contact subscribes.
+				 *
+				 * @param string              $provider The provider name.
+				 * @param array               $contact  {
+				 *    Contact information.
+				 *
+				 *    @type string   $email    Contact email address.
+				 *    @type string   $name     Contact name. Optional.
+				 *    @type string[] $metadata Contact additional metadata. Optional.
+				 * }
+				 * @param string[]|false      $lists    Array of list IDs to subscribe the contact to.
+				 * @param array|WP_Error      $result   Array with data if the contact was added or error if failed.
+				 * @param bool|null           $is_updating Whether the contact is being updated. If false, the contact is being created.
+				 */
+				do_action( 'newspack_newsletters_contact_subscribed', $provider->service, $contact, $lists, $result, $is_updating );
```

Comment:
The hook is triggered again here to handle the successful async subscription. I take suggestions to avoid duplication. Maybe a `finish_subscription` method?